File: src/main/java/in/dragonbra/javasteam/steam/handlers/steammatchmaking/callback/UserLeftLobbyCallback.java
package in.dragonbra.javasteam.steam.handlers.steammatchmaking.callback;

import in.dragonbra.javasteam.steam.handlers.steammatchmaking.Lobby;
import in.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg;
import in.dragonbra.javasteam.types.SteamID;

/**
 * This callback is fired whenever Steam informs us a user has left a lobby.
 *
 * @author lossy
 * @since 2022-06-21
 */
public class UserLeftLobbyCallback extends CallbackMsg {

    /**
     * ID of the app the lobby belongs to.
     */
    private int appId;

    /**
     * The SteamID of the lobby that a member left.
     */
    private SteamID lobbySteamID;

    /**
     * The lobby member that left.
     */
    private Lobby.Member user;

    public UserLeftLobbyCallback(int appId, long lobbySteamID, Lobby.Member user) {
        this.appId = appId;
        this.lobbySteamID = new SteamID(lobbySteamID);
        this.user = user;
    }

    public int getAppId() {
        return appId;
    }

    public SteamID getLobbySteamID() {
        return lobbySteamID;
    }

    public Lobby.Member getUser() {
        return user;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/handlers/steammatchmaking/callback/UserLeftLobbyCallback.kt
package `in`.dragonbra.javasteam.steam.handlers.steammatchmaking.callback

import `in`.dragonbra.javasteam.steam.handlers.steammatchmaking.Member
import `in`.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg
import `in`.dragonbra.javasteam.types.SteamID

/**
 * This callback is fired whenever Steam informs us a user has left a lobby.
 *
 * @param appID ID of the app the lobby belongs to.
 * @param lobbySteamID The SteamID of the lobby that a member left.
 * @param user The lobby member that left.
 *
 * @author Lossy
 * @since 2025-05-21
 */
class UserLeftLobbyCallback(
    val appID: Int,
    val lobbySteamID: SteamID,
    val user: Member,
) : CallbackMsg() {
    // No job set declared in SK.
}
